.github/workflows/feature-branch-prerelease.yml

```diff
@@ -0,0 +1,114 @@
+        default: true
+
+jobs:
+  authorize:
+    name: Authorize
+    runs-on: ubuntu-latest
+    steps:
+      - name: ${{ github.actor }} permission check to do a release
+        uses: 'lannonbr/repo-permission-check-action@2.0.2'
+        with:
+          permission: 'write'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  deploy:
+    name: Deploy
+    runs-on: ubuntu-latest
+    needs: [authorize]
+    permissions:
+      id-token: write
+      contents: write
+    env:
+      FEATURE_BRANCH: ${{ github.event.inputs.featureBranch }}
+      DRY_RUN: ${{ github.event.inputs.dryRun }}
+    strategy:
+      matrix:
+        node-version: [16.x]
+
+    steps:
+      - name: Check out git repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.FEATURE_BRANCH }}
+          fetch-depth: 0
+
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+      - name: Setup Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install project dependencies
+        run: |
+          yarn install --frozen-lockfile
+
+      - name: Build all packages
+        run: |
+          yarn build
+
+      - name: Test all packages
+        run: |
+          yarn test
+
+      - name: Lint all packages
+        run: |
+          yarn lint
+
+      - name: Configure Git User
+        run: |
+          git config --global user.name amplitude-sdk-bot
+          git config --global user.email amplitude-sdk-bot@users.noreply.github.com
+
+      - name: Configure NPM User
+        run: |
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_PUBLISH_TOKEN }}" > ~/.npmrc
+          npm whoami
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: arn:aws:iam::358203115967:role/github-actions-role
+          aws-region: us-west-2
+
+      # Converted to all lowercase and stripped of non-letter characters
+      - name: Transform feature branch name
+        run: |
+          echo "BRANCH_NAME=$(echo '${{ env.FEATURE_BRANCH }}' | tr -cd '[:lower:]')" >> $GITHUB_ENV
+
+     # Use --no-push to prevent pushing to remote
+      - name: Dry run pre-release version
+        if: ${{ env.DRY_RUN == true}}
+        run: |
+          GH_TOKEN=${{ secrets.GH_PUBLISH_TOKEN }} npm run deploy:version -- -y --no-private --conventional-prereleas --no-push --preid ${{ env.BRANCH_NAME}} --allow-branch ${{ env.FEATURE_BRANCH}}} --no-changelog
+
+      - name: Pre-release version
+        if: ${{ env.DRY_RUN == false }}
+        run: |
+          GH_TOKEN=${{ secrets.GH_PUBLISH_TOKEN }} npm run deploy:version -- -y --no-private --conventional-prereleas --preid ${{ env.BRANCH_NAME}} --create-release github
```

Comment:
```suggestion
          GH_TOKEN=${{ secrets.GH_PUBLISH_TOKEN }} npm run deploy:version -- -y --no-private --conventional-prerelease --preid ${{ env.BRANCH_NAME}} --create-release github
```

Reply:
I just found this too. Correct it https://github.com/amplitude/Amplitude-TypeScript/pull/621/commits/8737c29a010ddd49252e059549dd1de7c264d202